File: crates/app/src/types/modes.rs
// Ported from ./references/lazygit-master/pkg/gui/types/modes.go

pub struct Modes {
    pub filtering: FilteringMode,
    pub cherry_picking: Option<CherryPickingMode>,
    pub diffing: DiffingMode,
    pub marked_base_commit: MarkedBaseCommitMode,
}

pub struct FilteringMode;
pub struct CherryPickingMode;
pub struct DiffingMode;
pub struct MarkedBaseCommitMode;

File: pkg/gui/types/modes.go
package types

import (
	"github.com/quangdang46/slg/pkg/gui/modes/cherrypicking"
	"github.com/quangdang46/slg/pkg/gui/modes/diffing"
	"github.com/quangdang46/slg/pkg/gui/modes/filtering"
	"github.com/quangdang46/slg/pkg/gui/modes/marked_base_commit"
)

type Modes struct {
	Filtering        filtering.Filtering
	CherryPicking    *cherrypicking.CherryPicking
	Diffing          diffing.Diffing
	MarkedBaseCommit marked_base_commit.MarkedBaseCommit
}
